composer/metrics/nlp.py

```diff
@@ -266,9 +298,7 @@ def replace_underscore(text: str) -> str:
 
         return white_space_fix(remove_articles(handle_punc(lower(replace_underscore(answer))))).strip()
 
-    def update(self, outputs: List[str], labels: List[List[str]], batch: Optional[Dict[str, Any]] = None):
-        if batch is None:
-            batch = {}
+    def update(self, outputs: List[str], labels: List[List[str]], batch: Dict[str, Any]):
```

Comment:
Is the batch not optional? it seems optional cause everything accessed on it is defaulted, and you don't have to specify any of those things

Reply:
Only place it's called is in our original update_metric function which will always pass it a batch. Based on the code though, it's not optional (can't call .get on None)
I will note that this is a prime example jamming stuff in the batch that really is used in metric computation.

Reply:
well it was optional cause it used to do
```
if batch is None:
    batch = {}
```

Reply:
we decided to fix it in a later release